### PR TITLE
Add more CSP directives

### LINKED ISSUE #336: Enhance the CSP config with more CSP 2.0 goodness
We should lock down as many rules as possible and pay special attention to those that don't have a fall-back and are therefore wide open.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #336

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -54,21 +54,39 @@ module.exports = {
...
+      defaultSrc: ["'none'"],
+      baseUri: ["'self'"],
+      childSrc: ["'none'"],
       connectSrc: [
         "'self'",
         apiHost,
       ],
+      formAction: ["'self'"],
+      frameSrc: ["'none'"],
+      frameAncestors: ["'none'"],
```
Comment (muffinresearch) [inline config/default.js]: This is locking down what can be a frame ancestor. We'll need to configure this for the disco pane later on.

File: config/default.js [hunk 2]
```diff
@@ -54,21 +54,39 @@ module.exports = {
...
-      defaultSrc: ["'self'"],
+      defaultSrc: ["'none'"],
+      baseUri: ["'self'"],
+      childSrc: ["'none'"],
       connectSrc: [
         "'self'",
         apiHost,
       ],
+      formAction: ["'self'"],
+      frameSrc: ["'none'"],
```
Comment (muffinresearch) [inline config/default.js]: This is deprecated but I'm adding it for backwards compat.

**Comment by muffinresearch** [comment]
I realize I should add some comments to the code.

**Comment by mstriemer** [comment]
I'm guessing we don't need to set `plugin-types` because `object-src` is `"none"`?

**Comment by muffinresearch** [comment]
Gecko doesn't support `plugin-types`. But we could consider adding that at some point where it might be useful non-gecko UAs.

**Comment by mstriemer** [comment]
Looks good, as this grows I wonder if it's worth adding a helper to set the values so each environment can just have:

```js
CSP: cspConfig({apiHost, amoCDN}),
```

Looks good though, r+

**Pull request merged**